Test/XLTestOGlyco.cs

```diff
@@ -522,5 +519,56 @@ public static void OGlycoTest_DissociationTypeContainETD()
 
             Assert.That(isETDType);
         }
+
+        [Test]
+        [TestCase(true, true, false)] //write decoys and contaminants, database is targets
+        [TestCase(true, true, true)] //write decoys and contaminants, database is contaminants
+        [TestCase(false, false, false)] //do NOT write decoys and contaminants, database is targets
+        public static void OGlycoTest_ProteinInference(bool writeContaminants, bool writeDecoys, bool isContaminant)
+        {
+            string outputFolder = Path.Combine(TestContext.CurrentContext.TestDirectory, @"TESTGlycoData");
+            string proteinDatabase = Path.Combine(TestContext.CurrentContext.TestDirectory, @"GlycoTestData\N_O_glycoWithFileSpecific\\FourMucins_NoSigPeps_FASTA.fasta");
+            string spectraFileDirctory = Path.Combine(TestContext.CurrentContext.TestDirectory, @"GlycoTestData\N_O_glycoWithFileSpecific");
+            List<string> rawFilePaths = Directory.GetFiles(spectraFileDirctory).Where(p => p.Contains("mzML")).ToList();
+
+            // run task
+            CommonParameters commonParameters = new(dissociationType: DissociationType.HCD, ms2childScanDissociationType: DissociationType.EThcD);
+
+            Directory.CreateDirectory(outputFolder);
+            var glycoSearchTask = new GlycoSearchTask()
+            {
+                CommonParameters = commonParameters,
+                _glycoSearchParameters = new GlycoSearchParameters()
+                {
+                    OGlycanDatabasefile = "OGlycan.gdb",
+                    NGlycanDatabasefile = "NGlycan.gdb",
+                    GlycoSearchType = GlycoSearchType.OGlycanSearch,
+                    OxoniumIonFilt = true,
+                    DecoyType = DecoyType.Reverse,
+                    GlycoSearchTopNum = 50,
+                    MaximumOGlycanAllowed = 4,
+                    DoParsimony = true,
+                    WriteContaminants = writeContaminants,
+                    WriteDecoys = writeDecoys
+                }
+            };
+            glycoSearchTask.RunTask(outputFolder, new List<DbForTask> { new DbForTask(proteinDatabase, isContaminant) }, rawFilePaths, "");
+
+            List<string> expectedOutput = new()
```

Comment:
Do you want to do any testing of the actual contents of the files, instead of only checking if they exist?